apps/explorer/lib/explorer/exchange_rates/source.ex

```diff
@@ -56,17 +57,118 @@ defmodule Explorer.ExchangeRates.Source do
+  # retrieve from the pairs <external platform token address> - <eon token address> from the environment
+  @token_address_pairs "TOKEN_ADDRESS_PAIRS_EXT_PLATFORM_EON"
+  def fetch_token_address_pairs_for_swap do
+    token_list = System.fetch_env!(@token_address_pairs)
+    parse_token_list(token_list)
+  end
+
+  defp parse_token_list(token_list) do
+    token_pairs =
+      token_list
+      |> String.split(",")
+      |> Enum.map(&parse_token_pair/1)
+
+    Map.new(token_pairs)
+  end
+
+  defp parse_token_pair(token_pair) do
+    [name, address] = String.split(token_pair, "_")
+    {name, address}
+  end
+
+  # --------------------------------------------------------------------------------
+  # swap the addresses
+  defp swap_addresses(result_map, address_mapping) do
+    Enum.reduce(address_mapping, %{}, fn {old_address, new_address}, acc ->
+      swap_addresses(result_map, old_address, new_address, acc)
+    end)
+  end
+
+  defp swap_addresses(result_map, old_address, new_address, acc) do
+    old_value = Map.get(result_map, old_address)
+    updated_value = remove_zero_market_cap(old_value)
+    Map.put(acc, new_address, updated_value)
+  end
+
+  defp remove_zero_market_cap(map) do
+    case Map.get(map, "usd_market_cap") do
+      0.0 -> Map.delete(map, "usd_market_cap")
+      _ -> map
+    end
+  end
+
+  # --------------------------------------------------------------------------------
+  # retrieve the ZEN exchange rate from the database
+  @spec get_exchange_rate(String.t()) :: Token.t() | nil
+  defp get_exchange_rate(symbol) do
+    ExchangeRates.lookup(symbol)
+  end
+
+  # --------------------------------------------------------------------------------
+  # add the wrapped-zen map with the exchange rate read from the database
+  @wrapped_zen_env_var "WRAPPED_ZEN_ADDRESS"
+  defp create_wrapped_zen_map do
+    zen_exchange_rate = get_exchange_rate(Explorer.coin())
+    zen_usd_value =
+      case zen_exchange_rate do
+        %Explorer.ExchangeRates.Token{usd_value: uv} -> uv
+        _ -> nil
+      end
+
+    %{System.get_env(@wrapped_zen_env_var) => %{"usd" => zen_usd_value}}
+  end
+
+  # --------------------------------------------------------------------------------------------------------------------------------------------------------
+
+  @doc """
+  In the update_result_addresses method if the response processes id the one from the token_price api (v3/simple/token_price/<platform>) the addresses will
+  be swapped with their horizen-eon counterpart (using the TOKEN_ADDRESS_PAIRS_EXT_PLATFORM_EON environment variable)
+  Moreover the ZEN exchange rate is retrieved and an entry related to the wrapped-zen token is added to the result map
+  """
+  defp update_result_addresses(result, source_url) do
+
+    # check if the response is the one from the v3/simple/token_price/<platform> api
+    if String.contains?(source_url, "token_price") do
```

Comment:
I think this is ok, but it's a bit fragile.

Reply:
yes, modified checking a larger part of the path concatenate with the external platform value